### PR TITLE
More lsps

### PR DESCRIPTION
**Pull request by Veraticus**
Add some LSPs I personally wanted :)

### CODE REVIEW

File: plugins/nvim-lsp/options/servers.nix [hunk 1]
```diff
@@ -53,6 +53,7 @@ let
     };
     gopls = {
       languages = "Go";
+      packages = [ gopls ];
```
Comment (jooooscha) [inline plugins/nvim-lsp/options/servers.nix:56]: Why did you add this?
When packages is not set, it uses the name, which is fine in this case.
Also, just `gopls` is undefined; it should be `pkgs.gopls`.

File: plugins/nvim-lsp/options/servers.nix [hunk 2]
```diff
@@ -75,6 +76,10 @@ let
       languages = "text files";
       packages = [ ltex-ls ];
     };
+    lua_ls = {
+      languages = "Lua";
+      packages = [ ];
```
Comment (jooooscha) [inline plugins/nvim-lsp/options/servers.nix:81]: Does this LSP not require any additional packages?

File: plugins/nvim-lsp/options/servers.nix (hunk 1 above)
Comment (Veraticus) [reply plugins/nvim-lsp/options/servers.nix:56]: I don't remember >.>

File: plugins/nvim-lsp/options/servers.nix (hunk 2 above)
Comment (Veraticus) [reply plugins/nvim-lsp/options/servers.nix:81]: The package is `lua-language-server`.

Comment (jooooscha) [reply plugins/nvim-lsp/options/servers.nix:81]: I looked it up in [mason.nvim](https://mason-registry.dev/registry/list) There it is called lua-language-server. I think we should match that name.

Please add the package, and add a `warnIf (state <= 2211) "lua-language-server requries 23.05 or newer"`between `lua-language-server = ` and the `{`.

Thanks

**Comment by jooooscha** [comment]
replaced by c9b1447962371356a00bf93c5e331f2557ebd2a1
